freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/FreeIpaPostInstallService.java

```diff
@@ -57,5 +59,12 @@ public void postInstallFreeIpa(Long stackId) throws Exception {
             freeIpaClient.setUsernameLength(MAX_USERNAME_LENGTH);
         }
         passwordPolicyService.updatePasswordPolicy(freeIpaClient);
+
+        // CDPCP-591 - Enable User Sync after environment created
+        Set<String> environmentCrnFilter = new HashSet<>();
+        environmentCrnFilter.add(stack.getEnvironmentCrn());
+        userService.synchronizeUsers(
```

Comment:
env filter could be simply `Set.of(stack.getEnvironmentCrn())` and getting the usercrn would make more sense to be moved out

Reply:
I think that @lacikaaa was suggesting that you are attempting to add stack.getEnvironmentCrn() twice to environmentCrnFilter. You can simplify this block by
```
userService.synchronizeUsers(
    threadBasedUserCrnProvider.getAccountId(), 
    CrnService.checkUserCrn(threadBasedUserCrnProvider.getUserCrn()),
    Set.of(stack.getEnvironmentCrn(), Set.of(), Set.of());
```
It's strange to have to get the crn from the threadBasedUserCrnProvider and then call checkUserCrn. I think that we can change CrnService.checkUserCrn to CrnService.getUserCrn and not inject the ThreadBasedUserCrnProvider into this class at all.